### PR TITLE
NAS-116414 / 22.12 / ZFS Info Card and Tree Node Groups

### PR DESCRIPTION
**Pull request by denysbutenko**
Work in progress. Appreciate feedback.

Preview:
ZFS Info Card:
image

 Tree Node Groups:
image

image

### CODE REVIEW

File: src/app/interfaces/storage.interface.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export interface VDev {
   stats: VDevStats;
   device?: string;
   disk?: string;
+  name?: string;
```
Comment (undsoft) [inline src/app/interfaces/storage.interface.ts:20]: I would prefer to keep backend interfaces clean and add extra fields to a new interface.

**Review by undsoft — COMMENTED** [review]
Don't think we need to show empty groups like Log, Cache, etc.

File: src/app/interfaces/storage.interface.ts (hunk 1 above)
Comment (denysbutenko) [reply src/app/interfaces/storage.interface.ts:20]: When VDev has type VDevType.Mirror then name property is present in the API response

**Comment by denysbutenko** [comment]
New PR is #6855

**Pull request closed without merging**